Support read only & report forms on the patient sidebar

nmajor25 opened the pull request:
Shortcut Story ID: [sc-60600]


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a view-only modal form with a clear "Done" action, providing a distinct, non-editable interface for users.
	- Enhanced patient sidebar functionality with a new read-only form and associated modal widgets.

- **Tests**
	- Expanded testing to validate that the view-only and report forms display correctly with disabled inputs and appropriate footer actions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nmajor25 committed:
Support read only & report forms on the patient sidebar

nmajor25 commented on test/integration/patients/patient/sidebar.js:
```diff
@@ -386,6 +434,76 @@ context('patient sidebar', function() {
+        fx.data = testReadOnlyForm;
+
+        return fx;
+      });
```
This doesn't work and causes the test to fail.

The first `cy.routeForm()` in this test loads a script reducer form. And then here I want to override that and route the `testReadOnlyForm`.

We can do that with the other intercepts like `routeAction`, `routeFlow`, etc, but it doesn't seem to work for `routeForm`. So still trying to figure that out.

paulfalgout replied:
I believe all of the forms are bootstrapped.  We only load them individually when rendering a single form.

paulfalgout committed:
Remove report form handling for modals

Although report forms are often read only, the flag specifically is just about requesting data from another source.

paulfalgout committed:
Make routeForm unique for parallel calls
